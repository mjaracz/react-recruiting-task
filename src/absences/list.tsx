import React, {FC} from 'react'
import {useStyle} from './styled'
import {useList} from './hooks/useList'
import {CircularProgress, List} from '@material-ui/core'
import {AbsenceItem} from './item'

export const AbsencesList: FC = () => {
  const {absences} = useList()
  const {root} = useStyle()

  return (
    <div className={root}>
      {absences.isLoading
        ? <CircularProgress size={80}/>
        : (
          <List>
            {absences.list.map((item, index) => <AbsenceItem key={index} absences={item} />)}
          </List>
        )}
    </div>
  )
}