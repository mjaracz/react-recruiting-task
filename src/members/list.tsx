import React from 'react'
import {useList} from './hooks/useList'
import {useStyle} from '../absences/styled'
import {CircularProgress, List} from '@material-ui/core'
import {MemberItem} from './item'

export const MembersList = () => {
  const {members} = useList()
  const {root} = useStyle()
  return (
    <div className={root}>
      {members.isLoading
        ? <CircularProgress size={80}/>
        : (
          <List>
            {members.list.map((member, index) => <MemberItem key={index} member={member} />)}
          </List>
        )}
    </div>
  )
}