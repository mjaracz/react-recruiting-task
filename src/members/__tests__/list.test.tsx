import React from 'react'
import {render} from '@testing-library/react'
import {StoreProvider} from '../../storeProvider'
import {MembersList} from '../list'

describe('<AbsencesList/>', () => {
  it('should be defined and match snapshot', function () {
    const {baseElement} = render(<MembersList/>, { wrapper: StoreProvider })
    expect(baseElement).toMatchSnapshot()
  });
})