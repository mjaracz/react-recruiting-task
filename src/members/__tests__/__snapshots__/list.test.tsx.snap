// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AbsencesList/> should be defined and match snapshot 1`] = `
<body>
  <div>
    <div
      class="makeStyles-root-1"
    >
      <div
        class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
        role="progressbar"
        style="width: 80px; height: 80px;"
      >
        <svg
          class="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </div>
    </div>
  </div>
</body>
`;
